src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -619,7 +620,7 @@ public override Binder VisitEnumDeclaration(EnumDeclarationSyntax parent)
                 if (!binderCache.TryGetValue(key, out resultBinder))
                 {
                     Binder outer = VisitCore(parent.Parent); // a binder for the body of the type enclosing this type
-                    var container = ((NamespaceOrTypeSymbol)outer.ContainingMemberOrLambda).GetSourceTypeMember(parent.Identifier.ValueText, 0, SyntaxKind.EnumDeclaration, parent);
+                    var container = ((NamespaceOrTypeSymbol)ContainingMemberOrLambdaSkippingSimpleProgramEntryPoint(outer)).GetSourceTypeMember(parent.Identifier.ValueText, 0, SyntaxKind.EnumDeclaration, parent);
```

Comment:
Consider splitting this statement and the similar change above into multiple lines

Reply:
> Consider splitting this statement and the similar change above into multiple lines

In the next change (will go in as a separate PR) I am adjusting what binder we create in ```VisitCompilationUnit``` method, making the use of ```ContainingMemberOrLambdaSkippingSimpleProgramEntryPoint``` unnecessary. I am removing that helper and restoring call-sites to the state prior to this change.

---
In reply to: [381495525](https://github.com/dotnet/roslyn/pull/41706#discussion_r381495525) [](ancestors = 381495525)